### ISSUE TITLE
Skip links

### ISSUE THREAD

**Issue by marcus-herrmann** [owncloud/phoenix#2324]
# Why skip links are useful

> The main content is not usually the first thing on a web page. Keyboard and screen reader users generally must navigate a long list of navigation links, sub-lists of links, corporate icons, site searches, and other elements before ever arriving at the main content. This is particularly difficult for users with some forms of motor disabilities ([Source](https://webaim.org/techniques/skipnav/))

Imagine you are a keyboard only user, already know the overall structure of the website or web app, have recognized which items repeat on every page or view and you only want to navigate to the main content. Without a shortcut to the main content (that is a skip link) you have to tab through all focusable items of the header, the navigation and all other items coming before the main content. It is best practice (and also a guideline specified by the WCAG to offer means "for bypassing repetitive blocks of content". Skip links work best in conjunction with ARIA landmarks (`main`, `nav`, `[role="search"]`, `aside`, `footer` and so on) because landmarks in combination with a good headline structure and skip links offer screen reader users multiple ways of navigating to functionality or content directly (or giving them them the opportunity to get a quick overview of the website).

# How to implement them

To make most out of skip links, place them directly as the first element of the `<body>`. With this (and given there is no `tabindex` tinkering) they should be the first item of the tabindex (meaning: the first item that will get focus in the document once you hit the tab key).

Skip links don't have to be visible all the time, just on focus is fine. This can be solved in the following manner:

    a#skiptocontent {
    	position: absolute;
    	top: -40px;
    	left: 0px;
    }
    
    a#skiptocontent:focus {
    	top: 0;
    }

To use a visibility-hidden class instead of positioning the non-focused skip link off-canvas is also fine – as long as all the CSS properties that make it hidden in the first place (`position`, `clip`, `height`, `width`, `overflow` and so on) are reset **properly.**

Regarding wording, a simple "Skip to [target]" works best, so for example "Skip to main content". In my opinion it is better to call out to where the link interaction leads, instead of what exactly the links skips over.

While the link itself could be a component implementing one or more skip links means to, if necessary, add the `id` attribute to the skip targets – such such a component would not work in isolation.

# Good examples in the wild

Just visit the site and, before doing anything, hit your keyboards TAB key:

- [https://www.deque.com/](https://www.deque.com/)
- [https://developer.mozilla.org/](https://developer.mozilla.org/)
- [https://tatianamac.com/](https://tatianamac.com/)
- [https://www.gov.uk/](https://www.gov.uk/)

**Comment by marcus-herrmann** [comment]
I forgot that vue-router has issues with internal links like `<a href="#main">this</a>`. There is [issue #1668](https://github.com/vuejs/vue-router/issues/1668) but none of the workarounds listed there are actually moving focus to the container with the "#"-provided `id` (they are rather meddling with window.history or scrolling things into visible viewport, both of which hasn't reliable influence on the keyboard focus). So I guess I'll go the way of programmatically setting focus with JavaScript.

**Comment by PVince81** [comment]
@marcus-herrmann anything left to do that wasn't in the PR or can we close ?

**Comment by marcus-herrmann** [comment]
@PVince81 This can be closed.
